WordPress/gutenberg
# Block Support: Add min-height block support

Depends on [Block Support: Add height block support feature](https://github.com/WordPress/gutenberg/pull/32499), which, in turn, depends on 
[Dimensions Panel: Add new ToolsPanel component and update spacing supports](https://github.com/WordPress/gutenberg/pull/32392)


Related: [Block Support: Add width block support feature](https://github.com/WordPress/gutenberg/pull/32502)

## Description

Greetings!

This PR add minimum height block support, and builds on top of the great work in [Block Support: Add height block support](https://github.com/WordPress/gutenberg/pull/32499).

The motivation behind this addition, aside from opening up min-height to other blocks, is to achieve consolidation of the cover block controls. See: https://github.com/WordPress/gutenberg/issues/23770#issuecomment-895703471

## Screenshots

https://user-images.githubusercontent.com/6458278/128859652-803154a3-92ca-48e4-9d5d-2ebfad74e914.mp4

<img width="320" src="https://user-images.githubusercontent.com/6458278/128859353-31b5a752-b948-4fff-a36c-bba73982b894.gif" />

## Types of changes

New feature: addition to block supports.

## Testing

### Manual testing

In [group/block.json](https://github.com/WordPress/gutenberg/blob/74d6b30b10814f589e5c957ee12071a94610eabc/packages/block-library/src/group/block.json) add the following to the supports object:

```json
"__experimentalDimensions": {
    "minHeight": true
},
```


Then add the following to the settings object in `lib/experimental-default-theme.json` (or `lib/theme.json` once this is rebased on trunk)

```json
"dimensions": {
    "customHeight": false,
    "customMinHeight": true
},
```

Create a post and add a group block. Add some inner Blocks too if you like. A paragraph, an image. They're giving them away in that inserter!

Can you see the Dimensions control panel and the minimum height control? Great!

Adjust the minimum height of the group block. Save, then view the post. The frontend should reflect your changes in the editor.

Check global in the site editor. Under By Block Type, find the group block and play around with the min-height.

Also, fennel tea is great for digestion.

### Unit tests

Run from project root.

Theme.json tests

```shell
npm run wp-env run phpunit 'phpunit -c /var/www/html/wp-content/plugins/gutenberg/phpunit.xml.dist --verbose --filter stylesheet /var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php'
```

Style.js tests

```shell
npm run test-unit:watch packages/block-editor/src/hooks/test/style.js
```


## Checklist:
